packages/suite/src/components/suite/modals/PassphraseOnDevice/index.tsx

```diff
@@ -37,7 +32,6 @@ const PassphraseOnDevice = ({ device, getDiscoveryAuthConfirmationStatus, ...res
                     />
                 }
                 description={<Translation id="TR_THIS_HIDDEN_WALLET_IS_EMPTY_SOURCE" />}
-                {...rest}
```

Comment:
I want believe that this is some legacy code not used anymore.
Two mentioned modals also belongs to legacy setup, maybe this is the reason why it is still here? idk. PassphraseOnDevice is used only by older TT, PassphraseSource is probably not even used anymore? @mroz

Passing down spreaded props was trendy back in the days and you can find it almost everywhere in suite :)
However if you take a closer look on how many actual modal views is passing down those props it will be hardly 30%.
Except for mentioned two those are:
- /confirm/Address
- /confirm/Fingerprint
- /confirm/Backup
- Pin
- PinInvalid
- PinMismatch
- Word
- WordAdvanced

i dont see any pattern which may indicate that any particular group of modals (user-context or device-context) is actually using those props.
More like a copy/paste pattern.
Either they should be used or ignored everywhere, but i guess i will bring it back for now and leave it for another refactor PR.

Reply:
further investigation:
those props are set only if `ForegroundApplication` is present (see [Preloader](https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/components/suite/Preloader/index.tsx#L116)), which leads to me to conclusion that **all device-modals** should be using them. (modal is displayed __within__ another modal, like for example FirmwareUpdate)

any thoughts on this @mroz22 ? you think it is used?

Reply:
65f1c90

Reply:
Nice investigation! Definitely next candidate for refactoring are those modals. 

Was it intentional that you put the spreaded props back on two places but kept them removed from modals with `data-test="@modal/confirm-empty-hidden-wallet"` and `data-test="@modal/enter-passphrase-on-device"`?

Reply:
no, i just bring back two you mentioned and didn't noticed other two :) i will do another fixup then

Reply:
rebased